boards/waveshare/rp2040_matrix/rp2040_matrix-pinctrl.dtsi

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2024 Jonas Berg
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <dt-bindings/pinctrl/rpi-pico-rp2040-pinctrl.h>
+
+&pinctrl {
+
+	adc_default: adc_default {
+		group1 {
+			pinmux = <ADC_CH0_P26>, <ADC_CH1_P27>, <ADC_CH2_P28>, <ADC_CH3_P29>;
+			input-enable;
+		};
+	};
+
+	spi0_default: spi0_default {
+		group1 {
+			pinmux = <SPI0_TX_P3>;
+		};
+
+		group2 {
+			pinmux = <SPI0_RX_P4>;
+			input-enable;
+		};
+
+		group3 {
+			pinmux = <SPI0_SCK_P6>;
+		};
```

Comment:
Same as
https://github.com/zephyrproject-rtos/zephyr/pull/95232#discussion_r2314927033

And is it intentional to not configure CSn?

Reply:
Done.
I have pinmuxed the UART, SPI and ADC pins the same as the Waveshare RP2040-Zero board, and it does not define which pin that is intended for SPI CS.